docs/design/2022-06-30-partition-table-batchwrite-suppport.md

```diff
@@ -0,0 +1,121 @@
+complicated. Considering the workload of this feature, we have decide to support them in the future.
+In this feature, we only focus on Range partitioning and Hash partitioning.
+
+## API changes
+
+There is no API change in this feature. Users use the DataSource API to execute INSERT, REPLACE and
+DELETE as normal.
+
+## Detailed design
+
+Since TiSpark supports INSERT, REPLACE and DELETE semantics with the physical table, the
+responsibility is simplified to transform the `TableInfo` of the partition table to the physical
+table so that we can reuse the existing API rather than bring more complexity into the code.
+
+### Generate the expression for partitioning
+
+The original type of expression is `String`, we need to use `TiParser` to parse the expression in
+order to evaluate the partitioning expression for the latter steps.
+
+- For the `Range` and `Range Column`  partitioning, it also needs to extract the less than parts
+  from 'partition p0 less than xx ... partition p1 less than ...'
+
+### Locate the partition for each row
+
+In this step, we need to get the physical tableId for each row.
+
+For the `Range` partitioning, the expression can be a column or indeed a function.
+
+- If the expression is a column, we can use the column's value to compare with the less than parts
+  to locate the partition.
+- If the expression is a function, we can evaluate the function's value and then compare with the
+  less than parts to locate the partition
+
+For the `Range Column` partitioning, the expression can only be a column, so we choose the first if
+branch above.
+
+For the `Hash` partitioning, the result of the expression must be an integer, it can be a column or
+indeed a function.
+We take the result of the expression and mod it with the number of partitions, then use the
+remainder as the array index of the partition.
+
+### Group the rows by partition and invoke write API
+
+Group the rows by partition to generate `TiBatchWriteTable` and then invoke write API to write the
+rows in on transactions.
+
+There is one problem that `TiBatchWriteTable` use the `TableInfo` as a physical table which is not
+appropriate for the partitioning scenario.
+We need to use a new type such as `PhysicalTable` to replace the `TableInfo` in the write-associated
+interface.
+
+- For the partition table, the `TableInfo` generates n `PhysicalTable`s, each of which is a physical
+  table for a partition.
+- For the common table, the `TableInfo` generates a single `PhysicalTable` which is the physical
+  table for the whole table.
+
+## Test Design
+
+- insert into Range partition table
+- insert into Range Column partition table
+- insert into Hash partition table
+- delete from Range partition table
+- delete from Range Column partition table
+- delete from Hash partition table
+- replace into Range partition table
+- replace into Range Column partition table
+- replace into Hash partition table
```

Comment:
better to add a test for
1. different types of columns that are used by partition
2. different expression(function) for Range Column partition and Hash partition

Reply:
The design has changed, only YEAR() is supported

Reply:
Also, add `Date` type for range column